src/api/worker/rest/RestClient.ts

```diff
@@ -62,8 +69,19 @@ export class RestClient {
 		if (this.suspensionHandler.isSuspended()) {
 			return this.suspensionHandler.deferRequest(() => this.request(path, method, options))
 		} else {
-			return new Promise((resolve, reject) => {
+			return new Promise(async (resolve, reject) => {
 				this.id++
+
+				if (options.blobAccessParams) {
```

Comment:
sorry for the unprompted comment but this really seem like the wrong level to do this. RestClient doesn't know much about our API (and should probably know even less if you ask me). RestClient should be more of a general purpose HTTP library I feel like.

Could we do one of:

 * make `queryParams` lazy (optionally)? like `queryParams?: Dict | lazyAsync<Dict>`
 * use `SuspensionBehavior.Throw` and handle it on the outside? It is anyway one or two places that should handle this

Reply:
Your comment is very welcome! We agree that this is the wrong level, this was just a prototype to see if the solution would work. It did! We will eventually go with one of the two options you mentioned.